pypesto/objective/priors.py

```diff
@@ -0,0 +1,345 @@
+                'density_fun': log_f,
+                'density_dx': d_log_f_dx,
+                'density_ddx': dd_log_f_ddx}
+
+    elif parameter_scale == 'log':
+
+        def log_f_log(x_log):
+            """log-prior for log-parameters"""
+            return log_f(np.exp(x_log))
+
+        def d_log_f_log(x_log):
+            """derivative of log-prior w.r.t. log-parameters"""
+            return d_log_f_dx(np.exp(x_log)) * np.exp(x_log)
+
+        def dd_log_f_log(x_log):
+            """second derivative of log-prior w.r.t. log-parameters"""
+            return np.exp(x_log) * \
+                (d_log_f_dx(np.exp(x_log)) +
+                    np.exp(x_log) * dd_log_f_ddx(np.exp(x_log)))
+
+        return {'index': index,
+                'density_fun': log_f_log,
+                'density_dx': d_log_f_log,
+                'density_ddx': dd_log_f_log}
+
+    elif parameter_scale == 'log10':
+
+        log10 = np.log(10)
+
+        def log_f_log10(x_log10):
+            """log-prior for log10-parameters"""
+            return log_f(10**x_log10)
+
+        def d_log_f_log10(x_log10):
+            """derivative of log-prior w.r.t. log10-parameters"""
+            return d_log_f_dx(10**x_log10) * log10 * 10**x_log10
+
+        def dd_log_f_log10(x_log10):
+            """second derivative of log-prior w.r.t. log10-parameters"""
+            return log10**2 * 10**x_log10 * \
+                (dd_log_f_ddx(10**x_log10) * 10**x_log10
+                    + d_log_f_dx(10**x_log10))
+
+        return {'index': index,
+                'density_fun': log_f_log10,
+                'density_dx': d_log_f_log10,
+                'density_ddx': dd_log_f_log10}
+
+    else:
+        raise ValueError(f"NegLogPriors in parameters in scale "
+                         f"{parameter_scale} are currently not supported.")
+
+
+def _prior_densities(prior_type: str,
+                     prior_parameters: np.array) -> [Callable,
+                                                     Callable,
+                                                     Callable]:
+    """
+    Returns a tuple of Callables of the (log-)density (in untransformed =
+    linear scale), together with their first + second derivative
+    (= sensis) w.r.t. x.
+
+
+    Currently the following distributions are supported:
+
+    * uniform:
+        prior_parameters[0] and prior_parameters[1] give the lower and upper
+        boundary.
+    * normal:
+        normal distribution, with mean prior_parameters[0] and
+        standard deviation prior_parameters[1]
+    * laplace:
+        laplace distribution, with location prior_parameters[0] and
+        scale prior_parameters[1]
+    * logNormal:
+        logNormal distribution, where prior_parameters are mean and
+        standard deviation of the exp(X).
+
+    Currently not supported, but eventually planed are the
+    following distributions:
+
+    * logUniform
+    * logLaplace
+
+    """
+
+    if prior_type == 'uniform':
```

Comment:
Same as above, use I think the complexity and maintanance could be substantially improved by defining the individual prior functions once and then just using `sympy.diff` and `sympy.lambdify(...,modules=['numpy'])`. Add assumptions where necessary.

Reply:
here symbolic expressions would make more sense as more comlpicated, but as said fine for me for the moment.